app/scripts/router.js

```diff
@@ -244,9 +247,18 @@ function (
       // rendered for any reason, including if the view was
       // automatically redirected.
       var self = this;
-
       viewToShow.logScreen();
-      return viewToShow.render()
+      // perform any actions needed by the broker before the view has rendered
+      return self.broker.beforeRender()
+        .then(
+          function () {
+            return viewToShow.render();
+          }, function () {
+            // log the timeout event, this even should only happen in Firefox <43
+            viewToShow.logScreenEvent('broker.timeout');
```

Comment:
Might there be errors other than timeouts here?  It may be worth passing an explicit 'timeout' error param for a little future-proofing.